src/test/java/com/crud/tasks/controller/TaskControllerTestSuite.java

```diff
@@ -0,0 +1,123 @@
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(TrelloController.class)
+public class TaskControllerTestSuite {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    TaskController taskController;
+
+    @MockBean
+    TrelloFacade trelloFacade;
+
+
+    private TaskDto initTaskDto(Long number) {
+        return new TaskDto(number, "test" + number, "test content" + number);
+    }
+
+    private List<TaskDto> initTasksDtoList() {
+        List<TaskDto> tasks = new ArrayList<>();
+        tasks.add(0, initTaskDto(1L));
+        tasks.add(1, initTaskDto(2L));
+        return tasks;
+    }
+
+    @Test
+    public void shouldReturnAllTasksFromTheList() throws Exception {
+        //Given
+        List<TaskDto> tasks = initTasksDtoList();
+        when(taskController.getTasks()).thenReturn(tasks);
+        //When & Then
+        mockMvc.perform(get("/v1/task/getTasks").contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$", hasSize(2)))
+                .andExpect(jsonPath("$[0].id", is(1)))
+                .andExpect(jsonPath("$[0].title", is("test1")))
+                .andExpect(jsonPath("$[0].content", is("test content1")))
+                .andExpect(jsonPath("$[1].id", is(2)))
+                .andExpect(jsonPath("$[1].title", is("test2")))
+                .andExpect(jsonPath("$[1].content", is("test content2")));
+    }
+
+    @Test
+    public void shouldReturnSpecifiedTaskFromTheList() throws Exception {
+        //Given
+        List<TaskDto> tasks = initTasksDtoList();
+        when(taskController.getTask(1L)).thenReturn(tasks.get(0));
+        //When & Then
+        mockMvc.perform(get("/v1/task/getTask?taskId=1").contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id", is(1)))
+                .andExpect(jsonPath("$.title", is("test1")))
+                .andExpect(jsonPath("$.content", is("test content1")));
+
+    }
+
+    @Test
+    public void shouldDeleteSpecifiedTaskFromTheList() throws Exception {
+        //Given
+        when(taskController.deleteTask(1L)).thenReturn(true);
+        //When & Then
+        mockMvc.perform(delete("/v1/task/deleteTask?taskId=1").contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$", is(true)));
+    }
+
+    @Test
+    public void shouldPostTaskInToTheList() throws Exception {
+        //Given
+        Gson gSon = new Gson();
+        TaskDto createdTask = initTaskDto(1L);
+        doNothing().when(taskController).createTask(createdTask);
+        String jSonContent = gSon.toJson(createdTask);
+        //When & Then
+        mockMvc.perform(post("/v1/task/createTask")
```

Comment:
hmmm, czy na pewno chcemy doNothing, gdy tworzymy taska? nie chcemy zapisywac do bazy danych, ale chcielibysmy zwrocic to, co zwraca metoda kontrollera.